app/policies/profile_policy.rb

```diff
@@ -0,0 +1,23 @@
+class ProfilePolicy < ApplicationPolicy
+  def show?
+    user.present? || !record.hidden
```

Comment:
You probably wanted to do something like
```
(profile publicly available?) OR (user's own profile?)
```
Because right now any authenticated user can see any profile, even the closed ones.